aiutils/tftools/batch_normalizer.py

```diff
@@ -0,0 +1,94 @@
+"""A helper class for managing batch normalization state.
+
+This class is designed to simplify adding batch normalization
+(http://arxiv.org/pdf/1502.03167v3.pdf) to your model by
+managing the state variables associated with it.
+
+Important use note:  The function get_assigner() returns
+an op that must be executed to save the updated state.
+A suggested way to do this is to make execution of the
+model optimizer force it, e.g., by:
+
+  update_assignments = tf.group(bn1.get_assigner(),
```

Comment:
So, basically what it's saying is

```
opt = trainer.minimize(loss)
with tf.control_dependencies([opt]):
  new_opt = tf.group(<all batch norm updates>)
```

So, new_opt is actually the batch norm updates, but it calls the minimizer when you run it.